dbt/adapters/base/relation.py

```diff
@@ -36,6 +37,30 @@
 SerializableIterable = Union[Tuple, FrozenSet]
 
 
+@dataclass(frozen=True, eq=False, repr=False)
+class EventTimeFilter(FakeAPIObject, Hashable):
+    field_name: str
+    start: Optional[datetime] = None
+    end: Optional[datetime] = None
+
+    def render(self) -> str:
```

Comment:
Discussed w @mikealfare live today: we might want to move this method into the `BaseRelation` for ease of overwriting by adapter maintainers.